src/java/org/apache/cassandra/config/YamlConfigurationLoader.java

```diff
@@ -137,11 +143,49 @@ public Config loadConfig(URL url) throws ConfigurationException
         }
         catch (YAMLException e)
         {
-            throw new ConfigurationException("Invalid yaml: " + url + SystemUtils.LINE_SEPARATOR
-                                             +  " Error: " + e.getMessage(), false);
+            throw new ConfigurationException("Invalid yaml: " + url, e);
         }
     }
 
+    private static String readStorageConfig(URL url)
+    {
+        String content = "";
+
+        try
+        {
+            content = new String (Files.readAllBytes(Paths.get(String.valueOf(url).substring(5))));
+
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
```

Comment:
Should we throw a `ConfigurationException` out of this if we can't even read the config? Seems like we don't want to continue checking against an empty string?